Sources/Lifecycle/Lifecycle.swift

```diff
@@ -26,22 +26,32 @@ import Logging
 /// Represents an item that can be started and shut down
 public protocol LifecycleTask {
     var label: String { get }
+    var shutdownIfNotStarted: Bool { get }
     func start(_ callback: @escaping (Error?) -> Void)
     func shutdown(_ callback: @escaping (Error?) -> Void)
 }
 
+extension LifecycleTask {
+    public var shutdownIfNotStarted: Bool {
+        return false
+    }
+}
+
 // MARK: - LifecycleHandler
 
 /// Supported startup and shutdown method styles
 public struct LifecycleHandler {
     private let body: (@escaping (Error?) -> Void) -> Void
+    internal let noop: Bool
 
     /// Initialize a `LifecycleHandler` based on a completion handler.
     ///
     /// - parameters:
     ///    - callback: the underlying completion handler
-    public init(_ callback: @escaping (@escaping (Error?) -> Void) -> Void) {
+    ///    - noop: the underlying completion handler is a no-op
+    public init(_ callback: @escaping (@escaping (Error?) -> Void) -> Void, noop: Bool = false) {
```

Comment:
Not sure I understand what `noop` does. It seems a bit unsafe to let user set this?

Reply:
good point may this private